src/commands/topics.ts

```diff
@@ -166,6 +168,71 @@ async function editTopicConfig(topic: KafkaTopic): Promise<void> {
   editConfigForm.onDidDispose(() => disposable.dispose());
 }
 
+async function produceMessageFromFile(topic: KafkaTopic) {
+  if (!topic) {
+    vscode.window.showErrorMessage("No topic selected.");
+    return;
+  }
+  const options: vscode.OpenDialogOptions = {
+    canSelectMany: false,
+    openLabel: "Select JSON file",
+    filters: {
+      "JSON files": ["json"],
+    },
+  };
+
+  const fileUri = await vscode.window.showOpenDialog(options);
+  if (fileUri && fileUri[0]) {
+    const filePath = fileUri[0].fsPath;
+    const message = JSON.parse(fs.readFileSync(filePath, "utf-8"));
+
+    if (!message.key || !message.value || message.headers === 0) {
+      vscode.window.showErrorMessage(`Message must have a key, value, and headers.`);
+    }
```

Comment:
What do you think about doing something like this to make sure `headers` can be used with `.map()`?
```suggestion
    if (!message.key || !message.value || !Array.isArray(message.headers) {
      vscode.window.showErrorMessage(`Message must have a key, value, and headers.`);
    }
```